test/rewards/AutoCompoundingRewards.ts

```diff
@@ -686,6 +734,85 @@ describe('AutoCompoundingRewards', () => {
                     }
                 });
 
+                describe('auto-process rewards', () => {
+                    const AUTO_PROCESS_REWARDS_COUNT = 3;
+
+                    const setups = [
+                        { tokenSymbol: TokenSymbol.ETH, initialUserStake: toWei(10_000), totalRewards: toWei(11_000) },
+                        { tokenSymbol: TokenSymbol.BNT, initialUserStake: toWei(20_000), totalRewards: toWei(12_000) },
+                        { tokenSymbol: TokenSymbol.TKN, initialUserStake: toWei(30_000), totalRewards: toWei(13_000) },
+                        { tokenSymbol: TokenSymbol.TKN1, initialUserStake: toWei(40_000), totalRewards: toWei(14_000) },
+                        { tokenSymbol: TokenSymbol.TKN2, initialUserStake: toWei(50_000), totalRewards: toWei(15_000) },
+                        { tokenSymbol: TokenSymbol.TKN3, initialUserStake: toWei(60_000), totalRewards: toWei(16_000) },
+                        { tokenSymbol: TokenSymbol.TKN4, initialUserStake: toWei(70_000), totalRewards: toWei(17_000) },
+                        { tokenSymbol: TokenSymbol.TKN5, initialUserStake: toWei(80_000), totalRewards: toWei(18_000) }
+                    ];
+
+                    const tokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+                    const poolTokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+
+                    beforeEach(async () => {
+                        for (const [index, setup] of setups.entries()) {
+                            ({ token: tokens[index], poolToken: poolTokens[index] } = await prepareSimplePool(
+                                new TokenData(setup.tokenSymbol),
+                                setup.initialUserStake,
+                                setup.totalRewards
+                            ));
+
+                            await createProgram(
+                                distributionType,
+                                autoCompoundingRewards,
+                                tokens[index].address,
+                                setup.totalRewards,
+                                START_TIME
+                            );
+                        }
+
+                        await autoCompoundingRewards.setAutoProcessRewardsCount(AUTO_PROCESS_REWARDS_COUNT);
+                    });
```

Comment:
Why is it required?

Reply:
A preliminary initialization before testing function `autoProcessRewards`.

Reply:
`autoCompoundingRewards` should be an initialized proxy, so this setting is unneeded, no? Did something break without using it

Reply:
No. The initial value of the counter is 1, and it verified in a different test group.
In this test group, we set the counter to 3, and then we run several tests based on that counter.

Reply:
Ah, gotcha. I thought it was the default